pkg/operator/controller/ingress/deployment.go

```diff
@@ -1486,17 +1459,33 @@ func deploymentConfigChanged(current, expected *appsv1.Deployment, defaultIngres
 
 // copyProbe copies probe parameters that the operator manages from probe a to
 // probe b.
-func copyProbe(a, b *corev1.Probe, defaultIngressController bool) {
+func copyProbe(a, b *corev1.Probe) {
 	if a == nil || b == nil {
 		return
 	}
 
-	timeoutSeconds := b.TimeoutSeconds
-	a.DeepCopyInto(b)
-	// Ignore updates to timeoutSeconds for non-default ingresscontrollers
-	// to allow users to modify that parameter.
-	if !defaultIngressController {
-		b.TimeoutSeconds = timeoutSeconds
+	if a.ProbeHandler.HTTPGet != nil {
+		b.ProbeHandler.HTTPGet = &corev1.HTTPGetAction{
+			Path: a.ProbeHandler.HTTPGet.Path,
+			Port: a.ProbeHandler.HTTPGet.Port,
+			Host: a.ProbeHandler.HTTPGet.Host,
+		}
+		if a.ProbeHandler.HTTPGet.Scheme != "HTTP" {
+			b.ProbeHandler.HTTPGet.Scheme = a.ProbeHandler.HTTPGet.Scheme
+		}
+	}
+
+	// Users are permitted to modify the timeout, so *don't* copy it.
```

Comment:
Is changing the startupProbe timeout intentionally permitted?

```sh
$ oc -n openshift-ingress patch deploy/router-default --type=strategic --patch='{"spec":{"template":{"spec":{"$setElementOrder/containers":[{"name":"router"}],"containers":[{"livenessProbe":{"timeoutSeconds":1101},"name":"router","readinessProbe":{"timeoutSeconds":1202},"startupProbe":{"timeoutSeconds":1203}}]}}}}'
Warning: would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "router" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "router" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "router" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "router" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
deployment.apps/router-default patched
```

Wait a while....

```sh
$ sleep 30
$ oc -n openshift-ingress get  deploy/router-default -o yaml | grep -A8 Probe:
        livenessProbe:
          failureThreshold: 3
          httpGet:
            path: /healthz
            port: 1936
            scheme: HTTP
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 1101
--
        readinessProbe:
          failureThreshold: 3
          httpGet:
            path: /healthz/ready
            port: 1936
            scheme: HTTP
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 1202
--
        startupProbe:
          failureThreshold: 120
          httpGet:
            path: /healthz/ready
            port: 1936
            scheme: HTTP
          periodSeconds: 1
          successThreshold: 1
          timeoutSeconds: 1203
```

Reply:
I intentionally wrote the code that way, but I didn't mention it in [the enhancement document](https://github.com/openshift/enhancements/blob/master/enhancements/ingress/tunable-router-kubelet-probes.md), and [RFE-1786](https://issues.redhat.com/browse/RFE-1796) doesn't request the capability.  I don't feel strongly about this; would you prefer I remove the capability to modify the startup probe timeout?

Reply:
No. Though it is an additional thing for us to check when debugging live issues.

/lgtm